src/tasks/AndroidAppBuilder/ApkBuilder.cs

```diff
@@ -381,7 +382,13 @@ public ApkBuilder(TaskLoggingHelper logger)
         cmakeLists = cmakeLists.Replace("%Defines%", defines.ToString());
 
         File.WriteAllText(Path.Combine(OutputDir, "CMakeLists.txt"), cmakeLists);
-        File.WriteAllText(Path.Combine(OutputDir, monodroidSource), Utils.GetEmbeddedResource(monodroidSource));
+
+        string monodroidContent = Utils.GetEmbeddedResource(monodroidSource);
+        if (IsCoreCLR)
+        {
+            monodroidContent = ReplaceMonodroidCoreClrPlaceholders(monodroidContent);
```

Comment:
Nit: This doesn't replace all placeholders, so consider renaming to ReplaceAppContextPlaceholders